src/Security/Group.php

```diff
@@ -515,18 +515,21 @@ public function validate(): ValidationResult
         // and require an admin to perform this change in case it does.
         // This prevents "sub-admin" users with group editing permissions to increase their privileges.
         if ($this->Parent()->exists() && !Permission::check('ADMIN')) {
-            $inheritedCodes = Permission::get()
-                ->filter('GroupID', $this->Parent()->collateAncestorIDs())
-                ->column('Code');
             $privilegedCodes = Permission::config()->get('privileged_permissions');
-            if (array_intersect($inheritedCodes ?? [], $privilegedCodes)) {
-                $result->addError(
-                    _t(
-                        'SilverStripe\\Security\\Group.HierarchyPermsError',
-                        'Can\'t assign parent group "{group}" with privileged permissions (requires ADMIN access)',
-                        ['group' => $this->Parent()->Title]
-                    )
-                );
+            if (!empty($privilegedCodes)) {
+                $inheritedCodes = Permission::get()->filter([
+                    'GroupID' => $this->Parent()->collateAncestorIDs(),
+                    'Code' => $privilegedCodes,
+                ]);
+                if ($inheritedCodes->exists()) {
```

Comment:
Two optimisations in this query:
1. using `exists()` is more efficient than fetching columns
2. filtering by the privileged permissions means if there's no privileged codes the database can just say "nope nothing there"

This functionality is covered by `GroupTest::testValidatesPrivilegeLevelOfParent()`

Reply:
This functionality is covered by `GroupTest::testValidatesPrivilegeLevelOfParent()`